### PR TITLE
Mapping functions between DateSymbolsV1, TimeSymbolsV1, and neo symbols

### PR DESCRIPTION
**Pull request by sffc**
Part of #3598

Depends on #4188
Depends on #4189

### CODE REVIEW

**Review by Manishearth — DISMISSED** [review]
would be cool if we could design this as a wrapper provider but this is fine

File: components/datetime/src/provider/neo/adapter.rs [hunk 1]
```diff
@@ -0,0 +1,601 @@
...
+    pub const FORMAT_SHRT: Subtag = subtag!("6");
+}
+
+fn single_aux_subtag<M: KeyedDataMarker>(locale: &DataLocale) -> Result<Subtag, DataError> {
+    #[allow(clippy::unwrap_used)]
+    locale
+        .get_aux()
+        .and_then(|aux| aux.iter().next())
+        .ok_or_else(|| {
+            DataError::custom("Expected aux key")
```
Comment (Manishearth) [inline components/datetime/src/provider/neo/adapter.rs:29]: nit (optional): potentially error when there are two subtags

File: components/datetime/src/provider/neo/adapter.rs [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+            STADLN_WIDE => payload.weekdays.stand_alone_wide(),
+            STADLN_NARW => payload.weekdays.stand_alone_narrow(),
+            STADLN_SHRT => payload.weekdays.stand_alone_short(),
+            _ => None,
+        };
+        if let Some(result) = result {
+            return Ok(result.into());
+        }
+        let result = match subtag {
+            STADLN_SHRT | FORMAT_SHRT => payload.weekdays.format.short.as_ref(),
```
Comment (Manishearth) [inline components/datetime/src/provider/neo/adapter.rs:95]: praise: nice work with the match statements, this is a decent flow

**Review by Manishearth — APPROVED** [review]

File: components/datetime/src/provider/neo/adapter.rs (hunk 1 above)
Comment (sffc) [reply components/datetime/src/provider/neo/adapter.rs:29]: https://github.com/unicode-org/icu4x/pull/4201

**Pull request merged**